onnx/test/shape_inference_test.py

```diff
@@ -72,6 +72,7 @@ def _compare_value_infos(self, vi_type: TypeProto, inferred_vi_type: TypeProto)
             assert inferred_vi_type.HasField('tensor_type')
             assert vi_type.tensor_type.HasField('elem_type')
             assert inferred_vi_type.tensor_type.HasField('elem_type')
+            assert vi_type.tensor_type.HasField('shape') == inferred_vi_type.tensor_type.HasField('shape')
```

Comment:
the for-loop below should also be protected by `if (vi_type.tensor_type.HasField('shape'):`

Reply:
Although I think even if a tensor_type doesn't have a shape, tensor_type.shape.dim will just be empty. Still, I think adding protection is safer. Added. Thanks!

Reply:
The problem is that `tensor_type.shape` has the side-effect of initializing `shape` to be an empty list (an unfortunate design choice of protobuf) if the object doesn't have a `shape` field. The checker method should not have such a side-effect.